File: frontend/src/api/monitor.ts
import { get, post } from './client'

export interface ResourceMetrics {
  cpu_percent: number
  memory_percent: number
  memory_used_gb: number
  memory_total_gb: number
  disk_percent: number
  disk_used_gb: number
  disk_total_gb: number
}

export interface ServiceStatus {
  name: string
  status: string
  response_ms: number | null
}

export interface ResponseHistoryPoint {
  service_name: string
  response_ms: number | null
  status: string
  collected_at: string
}

export interface AlertItem {
  id: number
  level: string
  service_name: string
  message: string
  resolved: boolean
  created_at: string
  resolved_at: string | null
}

export interface LLMStatsResponse {
  total_24h: number
  by_module: Record<string, {
    count: number
    total_prompt_tokens: number
    total_completion_tokens: number
    avg_latency_ms: number
  }>
}

export interface OntologyStatsResponse {
  total_entities: number
  by_type: Record<string, number>
}

export interface AgentActivityResponse {
  total_agents: number
  published_agents: number
  total_skills: number
}

export interface DashboardOverview {
  resources: ResourceMetrics
  services: ServiceStatus[]
  alerts: AlertItem[]
  llm_stats: LLMStatsResponse
  ontology_stats: OntologyStatsResponse
  agent_activity: AgentActivityResponse
}

export const monitorApi = {
  overview() {
    return get<DashboardOverview>('/monitor/overview')
  },
  resources() {
    return get<ResourceMetrics>('/monitor/resources')
  },
  services() {
    return get<ServiceStatus[]>('/monitor/services')
  },
  responseHistory(hours: number = 1, service?: string) {
    const params = new URLSearchParams({ hours: String(hours) })
    if (service) params.set('service', service)
    return get<ResponseHistoryPoint[]>(`/monitor/response-history?${params}`)
  },
  alerts(limit = 20, resolved?: boolean, level?: string) {
    const params = new URLSearchParams({ limit: String(limit) })
    if (resolved !== undefined) params.set('resolved', String(resolved))
    if (level) params.set('level', level)
    return get<AlertItem[]>(`/monitor/alerts?${params}`)
  },
  resolveAlert(id: number) {
    return post<AlertItem>(`/monitor/alerts/${id}/resolve`)
  },
  llmStats() {
    return get<LLMStatsResponse>('/monitor/llm-stats')
  },
  ontologyStats() {
    return get<OntologyStatsResponse>('/monitor/ontology-stats')
  },
  agentActivity() {
    return get<AgentActivityResponse>('/monitor/agent-activity')
  },
  systemInfo() {
    return get<Record<string, any>>('/monitor/system-info')
  },
}

File: backend/app/schemas/monitor.py
from pydantic import BaseModel


class ResourceMetrics(BaseModel):
    cpu_percent: float
    memory_percent: float
    memory_used_gb: float
    memory_total_gb: float
    disk_percent: float
    disk_used_gb: float
    disk_total_gb: float


class ServiceStatus(BaseModel):
    name: str
    status: str
    response_ms: float | None = None


class ResponseHistoryPoint(BaseModel):
    service_name: str
    response_ms: float | None
    status: str
    collected_at: str


class AlertItem(BaseModel):
    id: int
    level: str
    service_name: str
    message: str
    resolved: bool
    created_at: str
    resolved_at: str | None = None


class LLMStatsResponse(BaseModel):
    total_24h: int
    by_module: dict[str, dict]


class OntologyStatsResponse(BaseModel):
    total_entities: int
    by_type: dict[str, int]


class AgentActivityResponse(BaseModel):
    total_agents: int
    published_agents: int
    total_skills: int


class DashboardOverview(BaseModel):
    resources: ResourceMetrics
    services: list[ServiceStatus]
    alerts: list[AlertItem]
    llm_stats: LLMStatsResponse
    ontology_stats: OntologyStatsResponse
    agent_activity: AgentActivityResponse
